R/dependencies.R

```diff
@@ -852,6 +858,114 @@ update_with_direct_deps <- function(project = ".",
   )
 }
 
+#' Recursively discover and build, check and install
+#' internal dependencies
+#'
+#' It discovers all dependencies starting from the repositories,
+#' determines the installation order and then
+#' builds, checks and installs them in order.
+#'
+#' @md
+#' @inheritParams rec_checkout_internal_deps
+#' @param steps (`character` vector) subset of "build", "check", "install";
+#'   useful to skip checking for example
+#' @param rcmd_args (`list`) with names `build`, `check`,
+#'   `install` which are vectors that are passed as separate arguments
+#'   to the `R CMD` commands
+#' @param artifact_dir (`character`) directory where build
+#'   tarball and logs go to
+#' @export
+#' @examples
+#' \dontrun{
+#' build_check_install_repos(
+#'   list(list(repo = "insightsengineering/teal", host = "https://github.com")),
+#'   feature = "main",
+#'   direction = "upstream",
+#'   local_repos = data.frame(repo = "insightsengineering/teal",
+#'   host = "https://github.com", directory = "../scratch1/teal/",
+#'   stringsAsFactors = FALSE)
+#' )
+#' build_check_install_repos(
+#'   list(list(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   host = "https://code.roche.com")),
+#'   feature = "master",
+#'   direction = "upstream",
+#'   local_repos = data.frame(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   host = "https://code.roche.com",
+#'   directory = "../example_ecosystem/stageddeps.electricity/",
```

Comment:
Not to change right now but this feels a little clunky as a way of giving the information something like this seems easier for users (not saying it should be this - just something to think about)

```
build_check_install_repos(c( "x/y@https://github.com"="remote", 
                                             "x/y2@https://github.com"="../y@local")
                                         )
```

Reply:
later (on cleanup)